src/main/java/yahav/photomosaic/ImagePixels.java

```diff
@@ -106,17 +108,25 @@ public double colorDistance(Color firstColor, Color secondColor) {
         return difference;
     }
 
-    public void closestColorDifference() {
+    public List closestColorDifference() throws IOException {
+        List<Image> closestFiles = new ArrayList<>();
         double min = 1.0;
```

Comment:
This is a subtle bug, you need to reset `min` when you set `compare`.
Also you should set it to be `Double.MIN_VALUE`.